lib/new_relic/agent/method_tracer.rb

```diff
@@ -48,12 +48,19 @@ module Agent
 
     module MethodTracer
 
-      def self.included clazz
-        clazz.extend ClassMethods
+      def self.included target
+        target.extend ClassMethods
+        NewRelic::Agent.logger.warn \
+          "`include NewRelic::Agent::MethodTracer` is deprecated\n" \
+          "Please use `prepend NewRelic::Agent::MethodTracer` instead"
       end
```

Comment:
We may be able to change this to:
```ruby
      def self.included target
        target.prepend PrependMethods
        NewRelic::Agent.logger.warn \
          "`include NewRelic::Agent::MethodTracer` is deprecated\n" \
          "Please use `prepend NewRelic::Agent::MethodTracer` instead"
      end
```
which, if it works, means we can throw away `ClassMethods` altogether.  Users who `include` will still continue to work with a deprecation warning logged.